### PR TITLE
US023: apply - slot cannot be modified if it is in use

### PR DESCRIPTION
**Pull request by dilmurod-rozikov**

### CODE REVIEW

File: ParkingZoneApp.Tests/Controllers/Admin/ParkingSlotControllerUnitTests.cs [hunk 1]
```diff
@@ -223,7 +232,7 @@ public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnNotFoundI
         }
 
         [Fact]
-        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnModelError()
+        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnModelErrorIfNumberIsNotUnique()
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/Controllers/Admin/ParkingSlotControllerUnitTests.cs]:
```suggestion
        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenNumberIsNotUniqueAndModelStateIsFalseAndReturnsViewResult()
```

File: ParkingZoneApp.Tests/Controllers/Admin/ParkingSlotControllerUnitTests.cs [hunk 2]
```diff
@@ -246,13 +255,38 @@ public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnModelErro
                     .Verify(x => x.IsUniqueNumber(editVM.ParkingZoneId, editVM.Number), Times.Once);
         }
 
+        [Fact]
+        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenReturnModelErrorIfSlotIsInUse()
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/Controllers/Admin/ParkingSlotControllerUnitTests.cs]: why return it is grammatically wrong i think lets do like this 
```suggestion
        public void GivenEditVMAndParkingSlotId_WhenEditPostIsCalled_ThenSlotIsInUseAndModelStateIsFalseAndReturnsViewResult()
```

**Review by SamDotNetDev — CHANGES_REQUESTED** [review]

**Review by SamDotNetDev — APPROVED** [review]

**Comment by dilmurod-rozikov** [comment]
[-	US023 (1): Modify slot editing action. Slot category cannot be modified if it is in use. In this case make this field readonly and show corresponding warning message near it. Also check on POST if a slot is in use.](https://trello.com/c/NFyV3I7S/19-us023-1-modify-slot-editing-action-slot-category-cannot-be-modified-if-it-is-in-use-in-this-case-make-this-field-readonly-and-s)

File: ParkingZoneApp/Models/Entities/ParkingSlot.cs [hunk 3]
```diff
@@ -29,5 +29,11 @@ public class ParkingSlot
 
         [Required]
         public virtual ICollection<Reservation> Reservations { get; set; }
+
+        [NotMapped]
+        public bool IsSlotInUse
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Models/Entities/ParkingSlot.cs]:
```suggestion
        public bool IsInUse
```

Comment (abdullokhtolibjonov) [reply ParkingZoneApp/Models/Entities/ParkingSlot.cs]: we dont need word "Slot" because its staying in model called "Slot"

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**